client/components/domains/register-domain-step/index.jsx

```diff
@@ -314,7 +314,7 @@ const RegisterDomainStep = React.createClass( {
 						return callback();
 					}
 
-					if ( ! domain.match( /^[A-Za-z][A-Za-z0-9\.-]{2,}\.[A-Za-z]{2,}$/ ) ) {
+					if ( ! domain.match( /^([a-z0-9_]([a-z0-9_-]*[a-z0-9_])?\.)*[a-z0-9]([a-z0-9-]*[a-z0-9])?\.[a-z]{2,63}$/i ) ) {
```

Comment:
Good point - since we are only dealing with hostnames here, the underscore should not be allowed. Rebased (to pull in Node 6.9.0) and updated.